src/lib.rs

```diff
@@ -389,6 +393,7 @@ mod test {
     use std::io::Write;
     use std::net::TcpListener;
     use std::thread;
+    use pin_project::pin_project;
```

Comment:
It probably makes more sense just to use pin-project-lite here as well - it's only one type that needs to project.